README.md

```diff
@@ -8,29 +8,208 @@ JavaScript Reduce
 3. Practice building small pieces of code that can be used later
 
 ## Introduction
+In the world of programming, we often work with lists. Sometimes we want to transform elements in that list to another
+value — but other times, we want to **aggregate** a result. In other words, we want to _reduce_ a list to a single value —
+it could be a string, a number, a boolean, ...
 
-- Introduce students to reduce() with an example -- maybe something about taking a collection, repeating an action on it, and returning the accumulation of that action
+![Amazing parking job](http://www.carcrushing.com/wp-content/uploads/2014/11/Bad-Parking-Jobs-Car-Crushing-004.jpg)
 
-## Implementation
+For example, let's say our friend is really bad at parking. We have an entire stack of his parking fees in front of us.
+Wouldn't it be great if we could reduce all of those parking fees to a single total value? Then we can tell him how much
+he owes the city for his obnoxious behavior. Shame on him.
 
-- It might be a good idea to refresh students' memories on writing their own forEach()
-- There's no need to implement the full polyfill, and this `reduce()` should not be attached to the `Array` prototype.
-- Instead, students could end up with something as basic as:
+## Coupon Queen
+![Honey Boo Boo](https://media.giphy.com/media/ponvUa3urCwJq/giphy.gif)
 
-``` javascript
-function reduce(collection, fn, init) {
-  let value = init
+Let's pretend we're Honey Boo Boo. Bored from parading around in beauty pageants, we decide to take up another hobby —
+programming. This might even help out Mama June, the Coupon Queen! Pulling up some store data, we're
+interested in getting a total price of all products that have a discount of 50% or more. The store data looks like
+this:
 
-  for (const item in collection) {
-    value = fn(temp, collection[item], item, collection)
+```js
+const products = [
+  { name: 'Head & Shoulders Shampoo', standardPrice: 4.99, discount: .6 },
+  { name: 'Twinkies', standardPrice: 7.99, discount: .45 },
+  { name: 'Oreos', standardPrice: 6.49, discount: .8 },
+  { name: 'Jasmine-scented bath pearls', standardPrice: 13.99, discount: .7 },
+];
+```
+
+This means that we're essentially going to reduce the array of products to a _single value_ — in this case the total
+price of all products that qualify. Let's create a function that has an initial value, then iterates the given products
+and adds their price to the total price if its discount is higher than 50% (represented as `.5` in our code). When the
+loop has finished, we return our `totalPrice` result:
+
+```js
+function getTotalAmountForProducts(products) {
+  let totalPrice = 0;
+
+  products.forEach(function (product) {
```

Comment:
Total style nit, but should we use a fat-arrow function here? Both because it looks nice and because we can remind students how to use them (even if we get no benefit here from its lexical binding)?